CMakeLists.txt

```diff
@@ -103,17 +103,7 @@ if(WITH_MOAB)
     set(MOAB_ROOT "${DEPS_ROOT_DIR}")
   endif()
   MESSAGE("-- MOAB Root: ${MOAB_ROOT}")
-  find_package(MOAB)
-  message("-- MOAB Found: ${MOAB_FOUND}")
-  if(${MOAB_FOUND})
-    include_directories(${MOAB_INCLUDE_DIRS})
-    get_filename_component(MOAB_LIBRARY_DIRS ${MOAB_LIBRARY} PATH)
-    if(APPLE)
-      link_directories(${MOAB_LIBRARY_DIRS})
-    else(APPLE)
-      link_directories(${MOAB_LIBRARY_DIRS})
-    endif(APPLE)
-  endif(${MOAB_FOUND})
+  find_package(MOAB REQUIRED)
```

Comment:
MOAB is not required for PyNE

Reply:
Yes, I know. I have noticed there is a statement available `WITH_MOAB`. So, MOAB is only required when any user sets this value to `ON`. Please note, by default this is already set to `ON` in the configuration file. 

I think we should make this required here, thus user can gets error message if they have turned `ON` `WITH_MOAB` and they do not have MOAB installed. Please, let me know should I still remove this or not.

Another note: it seems that a pip package of MOAB will soon be available. I am currently adjusting different configurations to find MOAB in the [pyne/scikit-build-core](https://github.com/pyne/pyne/tree/scikit-build-core) branch. Once all the tests are passing, I will make a PR for the big update.

Reply:
I understand why to make this required now